src/deepsparse/pipelines/embedding_extraction.py

```diff
@@ -0,0 +1,298 @@
+        return self._extraction_strategy
+
+    @property
+    def return_numpy(self) -> bool:
+        """
+        :return: if True returns embeddings as numpy arrays, uses lists of=
+            floats otherwise
+        """
+        return self._return_numpy
+
+    @property
+    def input_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that inputs to this pipeline must comply to
+        """
+        return self.base_pipeline.input_schema
+
+    @property
+    def output_schema(self) -> Type[BaseModel]:
+        """
+        :return: pydantic model class that outputs of this pipeline must comply to
+        """
+        return EmbeddingExtractionOutput
+
+    def setup_onnx_file_path(self) -> str:
+        """
+        Performs setup done in pipeline parent class as well as truncating the
+        model to an intermediate layer for embedding extraction
+
+        :return: file path to the processed ONNX file for the engine to compile
+        """
+        onnx_path = self.base_pipeline.onnx_file_path
+
+        # TODO: use truncate_onnx_model, not truncate_transformer_onnx_model
+        #  potentially find a way to keep transformers specific helpers
+        #  basically want to copy the UX here so we can push
+        if self._emb_extraction_layer is not None:
+            (
+                onnx_path,
+                self.onnx_output_names,
+                self._temp_model_directory,
+            ) = truncate_transformer_onnx_model(
+                onnx_path,
+                emb_extraction_layer=self._emb_extraction_layer,
+                hidden_layer_size=self._model_size,
+            )
+        else:
+            _LOGGER.info("EmbeddingExtractionPipeline - skipping model truncation")
+
+        return onnx_path
+
+    def parse_inputs(self, *args, **kwargs) -> BaseModel:
+        """
+        :param args: ordered arguments to pipeline, either a input_schema object,
+            a string text, or a list of inputs
+        :param kwargs: keyword arguments to pipeline
+        :return: pipeline arguments parsed into the given `input_schema`
+            schema if necessary. If an instance of the `input_schema` is provided
+            it will be returned
+        """
+        return self.base_pipeline.parse_inputs(*args, **kwargs)
+
+    def process_inputs(self, inputs: BaseModel) -> List[numpy.ndarray]:
+        """
+        Tokenizes input
+
+        :param inputs: inputs to the pipeline.
+        :return: inputs of this model processed into a list of numpy arrays that
+            can be directly passed into the forward pass of the pipeline engine
+        """
+        return self.base_pipeline.process_inputs(inputs)
+
+    # TODO @rahul-tuli: this assumes extra information from transformers need to make
+    #  this generic
+    def process_engine_outputs(
+        self,
+        engine_outputs: List[numpy.ndarray],
+        pad_masks: numpy.ndarray,
+        cls_masks: numpy.ndarray,
```

Comment:
this will need to accept any generic args passed in from base pipeline

```suggestion
        **kwargs,
```